mgmt/rest/server.go

```diff
@@ -252,15 +252,15 @@ func (s *Server) run(addrString string) {
 		config := &tls.Config{Certificates: []tls.Certificate{cer}}
 		ln, err := tls.Listen("tcp", addrString, config)
 		if err != nil {
-			s.err <- err
+			log.Fatal(err)
 		}
 		s.serverListener = ln
 		s.wg.Add(1)
 		go s.serveTLS(ln)
 	} else {
 		ln, err := net.Listen("tcp", addrString)
 		if err != nil {
-			s.err <- err
```

Comment:
There is mechanism of monitoring errors [see here: monitorErrors](https://github.com/intelsdi-x/snap/blob/master/snapteld.go#L379) using `err channel`:
```
func monitorErrors(ch <-chan error) {
	// Block on the error channel. Will return exit status 1 for an error or just return if the channel closes.
	err, ok := <-ch
	if !ok {
		return
	}
	log.Fatal(err)
}
```
So just add "return" after lines 255 and 263 are enough to avoid panic, **but** after such change the logs are confusing - I mean log "REST started" appears as a first one ([see here](https://github.com/intelsdi-x/snap/blob/master/mgmt/rest/server.go#L214)) even it is not true.
```
INFO[2017-04-19T15:42:31+02:00] Starting REST API on :8181                    _module=_mgmt-rest
INFO[2017-04-19T15:42:31+02:00] REST started                                  _block=start _module=_mgmt-rest
INFO[2017-04-19T15:42:31+02:00] module started                                _module=snapteld block=main snap-module=REST
INFO[2017-04-19T15:42:31+02:00] setting plugin trust level to: disabled
INFO[2017-04-19T15:42:31+02:00] snapteld started
FATA[2017-04-19T15:42:31+02:00] listen tcp :8181: bind: address already in use
```
I just want to share my observations about this with you. Basically, I am ok with logging Fatal as soon as possible. I have just concerns if there are other similar issues and about what next with that existing `monitorErrors` - is it still need and should be kept or there is a plan to remove it.